code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java

```diff
@@ -560,6 +562,83 @@ static String getSharedStateEcid(final Map<String, Object> edgeIdentityState) {
         return DataReader.optString(ecidMap, MessagingConstants.SharedState.EdgeIdentity.ID, null);
     }
 
+    /**
+     * Determines if the provided push token should be synced to Adobe Journey Optimizer via an Edge
+     * network request.
+     *
+     * @param configSharedState A {@link Map} containing the configuration shared state
+     * @param newPushToken A {@code String} containing the push token to be synced
+     * @return {@code boolean} indicating if the push token should be synced
+     */
+    static boolean shouldSyncPushToken(
+            @Nullable final Map<String, Object> configSharedState,
+            @Nullable final String newPushToken) {
+        if (StringUtils.isNullOrEmpty(newPushToken)) {
+            Log.debug(
+                    MessagingConstants.LOG_TAG,
+                    "shouldSyncPushToken",
+                    "New push token is null or empty, push token will not be synced.");
+            return false;
+        }
+
+        final boolean isPushRegistrationPaused =
+                DataReader.optBoolean(
+                        configSharedState,
+                        MessagingConstants.SharedState.Configuration.PUSH_REGISTRATION_PAUSED,
+                        false);
+
+        if (isPushRegistrationPaused) {
+            Log.debug(
+                    MessagingConstants.LOG_TAG,
+                    "shouldSyncPushToken",
+                    "Push registration is paused, push token will not be synced.");
+            return false;
+        }
```

Comment:
The logic we want is not to simply pause all push requests, but to either sync on all requests or to sync only when the value changes.
I'd suggest

1. Change config flag to `messaging.pushForceSync`
2. Default for pushForceSync is false, meaning only sync on value change.
3. Update check on line 599 to `existingPushToken != null && existingPushToken.equals(newPushToken) && !pushForceSync)`

Reply:
i'll get this fixed